### ISSUE TITLE
[BUG] - Wrong answer to the correct expected output

### ISSUE THREAD

**Issue by Eleazer85** [LeetCode-Feedback/LeetCode-Feedback#21991]
### LeetCode Username

eel123

### Problem Number, Title, and Link

185. Department Top Three Salaries https://leetcode.com/problems/department-top-three-salaries/description/?lang=pythondata

### Bug Category

Incorrect test case *(Output of test case is incorrect as per the problem statement)*

### Bug Description

I ran my code and got the same output as the provided example. However, the test still failed, and the test case is identical to the example provided in the problem. The expected output doesn't seem correct. Can you help clarify this?

### Language Used for Code

Python/Python3

### Code used for Submit/Run operation

```bash
import pandas as pd
import numpy as np

def top_three_salaries(employee: pd.DataFrame, department_pd: pd.DataFrame) -> pd.DataFrame:
    employee.sort_values(by="departmentId",ascending=True,inplace=True)

    output_df = {
        "Department":[],
        "Employee":[],
        "Salary":[]
    }
    for idx , rows in department_pd.iterrows():
        department = employee.loc[employee["departmentId"] == rows["id"]]
        department.sort_values(by="salary",ascending=False,inplace=True)

        unique_salary = department['salary'].unique()
        unique_salary = np.sort(unique_salary)
        top_3 = unique_salary[-3:]

        result = department[department["salary"].isin(top_3)]
        
        for idx , rows in result.iterrows(): 
            dp_name = department_pd[department_pd["id"] == rows["departmentId"]]
            output_df["Department"].append(dp_name["name"]) #This is department name
            output_df["Employee"].append(rows["name"])
            output_df["Salary"].append(rows["salary"])

    return pd.DataFrame(output_df)
```

### Expected behavior

The expected behavior was that my code will fail even if the output is the same as the provided example.

### Screenshots

![Screenshot (349)](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/72891905/7c885cd2-e3c0-46a7-bc05-bbb504d4b3ee)

![Screenshot (351)](https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/72891905/32c284b1-b006-49da-8bdb-8af8a5f63db5)

### Additional context

As you can see from my screenshot, other people have the same output, but their cases are accepted. I even try to check my DataFrame column name, but everything seems to be correct.

**Comment by Eleazer85** [comment]
Sure, here's the specific test case that failed is as follows:

Test Case:
- Input:
  - Employee DataFrame:
    | id | name  | salary | departmentId |
    |----|-------|--------|--------------|
    | 1  | Joe   | 85000  | 1            |
    | 2  | Henry | 80000  | 2            |
    | 3  | Sam   | 60000  | 2            |
    | 4  | Max   | 90000  | 1            |
    | 5  | Janet | 69000  | 1            |
    | 6  | Randy | 85000  | 1            |
    | 7  | Will  | 70000  | 1            |

  - Department DataFrame:
    | id | name  |
    |----|-------|
    | 1  | IT    |
    | 2  | Sales |

- Expected Output:
  | Department | Employee | Salary |
  |------------|----------|--------|
  | IT         | Max      | 90000  |
  | IT         | Joe      | 85000  |
  | IT         | Randy    | 85000  |
  | IT         | Will     | 70000  |
  | Sales      | Henry    | 80000  |
  | Sales      | Sam      | 60000  |

- Actual Output:

  | Department | Employee | Salary |
  |------------|----------|--------|
  | IT         | Max      | 90000  |
  | IT         | Joe      | 85000  |
  | IT         | Randy    | 85000  |
  | IT         | Will     | 70000  |
  | Sales      | Henry    | 80000  |
  | Sales      | Sam      | 60000  |

As you can see, the actual output matches the expected output exactly.

`

> Winston Tang commented: Dear eel123,
> 
> Thank you for reaching out to us about your concerns with problem 185, Department Top Three Salaries. We appreciate your effort in detailing your code implementation and the steps you have taken to identify potential problems.
> 
> To further assist you, we would need more specific information. Could you please provide the distinct test case that failed and mentioned the exact expected and actual outputs? Knowing the specifics will help us verify whether there is an issue with the problem statement, test cases, or if the error lies within your implemented solution.
> 
> Also, please cross-verify that the problem's constraints and specifications are entirely met by your solution. We appreciate your patience and assure you that we aim to provide the best problem-solving experience for our users.
> 
> Best Regards, LeetCode Support Team

**Comment by Eleazer85** [comment]
Hey, I just found out why this give me a wrong answer. It's because the "department" output is expected to be a string, I append pandas Series instead into it. Here's line of the code that causes the problem
```
output_df["Department"].append(dp_name["name"]) #line 24
```
I just realized this when I check `dp_name["name"]` datatype and it says `<class 'pandas.core.series.Series'>`
I added `.iloc` to it so it only append the string inside of the pandas series. So here's the modified code in line 24: 
```
output_df["Department"].append(dp_name["name"].iloc[0])
```
